src/query.rs

```diff
@@ -252,28 +281,32 @@ mod tests {
                         )
                     };
 
-                assert!(
-                    run_query_on(&indexed_crate_new, &indexed_crate_new)
-                        .1
-                        .is_empty(),
-                    "{}",
-                    crate_pair
-                );
-                assert!(
-                    run_query_on(&indexed_crate_old, &indexed_crate_old)
-                        .1
-                        .is_empty(),
-                    "{}",
-                    crate_pair
-                );
+                let assert_no_false_positives_in_nonchanged_crate =
+                    |crate_: &VersionedIndexedCrate,
+                     crate_version: &str| {
```

Comment:
This code is getting quite difficult to read. This is already a second closure inside a closure inside the test function. Try to move some of this functionality out into functions of their own?

Reply:
Fixed in 9545e230ccfdd7260810d25f8999059d5fcbbe1b and 6898e90b3cef46952492dbd1882b71137bb68054